ddtrace/contrib/cassandra/session.py

```diff
@@ -31,32 +36,133 @@ def traced_connect(func, instance, args, kwargs):
     session = func(*args, **kwargs)
     if not isinstance(session.execute, wrapt.FunctionWrapper):
         # FIXME[matt] this should probably be private.
-        setattr(session, 'execute', wrapt.FunctionWrapper(session.execute, traced_execute))
+        setattr(session, 'execute_async', wrapt.FunctionWrapper(session.execute_async, traced_execute_async))
     return session
 
-def traced_execute(func, instance, args, kwargs):
-    cluster = getattr(instance, 'cluster', None)
+def _close_span_on_success(result, future):
+    span = getattr(future, CURRENT_SPAN, None)
```

Comment:
It worked very well in general, using the `Future` as a carrier, so it's safe to retrieve it that way

Reply:
So is it good that way or do you think I should change something here?

Reply:
yes! that one is a good approach! 👍